pkg/skaffold/docker/client.go

```diff
@@ -105,9 +106,32 @@ type ExitCoder interface {
 	ExitCode() int
 }
 
-// newMinikubeAPIClient returns a docker client using the environment variables
+// newMinikubeAPIClient returns a client using the environment variables
 // provided by minikube.
 func newMinikubeAPIClient(ctx context.Context, minikubeProfile string) ([]string, client.CommonAPIClient, error) {
+	driver, err := getMinikubeDriver(ctx, minikubeProfile)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if driver == "docker" {
+		return newMinikubeAPIClientWithDockerDriver(ctx, minikubeProfile)
+	}
+
+	cmd, err := cluster.GetClient().MinikubeExec(ctx, "image", "load")
+	if err != nil {
+		return nil, nil, fmt.Errorf("executing minikube command: %w", err)
```

Comment:
lets have actionable errors for there. Add an action code and a suggestion to see `minikube logs` to figure out what is wrong.

Reply:
Note: minikube also provides exit codes. See example here https://github.com/GoogleContainerTools/skaffold/pull/7904/files#diff-f7852592ac185659a3acd7d1b0c5127703270e8c7143128f79cab2d24c02011eR140

Maybe we should refactor the code in the above lines and make a func to extract error codes from minikube errors

Reply:
I think you added the wrong link here @tejal29

Reply:
https://github.com/GoogleContainerTools/skaffold/blob/eefc9fa01555d972847aeddd11511c929007165b/pkg/skaffold/docker/client.go They are in the same file. Just a bit below. I think you since you updated the diff, the link does not work.

Reply:
https://github.com/GoogleContainerTools/skaffold/blob/eefc9fa01555d972847aeddd11511c929007165b/pkg/skaffold/docker/client.go#L117